tableau_blueprints/refresh_datasource.py

```diff
@@ -0,0 +1,105 @@
+EXIT_CODE_UNKNOWN_ERROR = 3
+EXIT_CODE_INVALID_CREDENTIALS = 200
+EXIT_CODE_INVALID_RESOURCE = 201
+EXIT_CODE_FINAL_STATUS_ERRORED = 211
+EXIT_CODE_FINAL_STATUS_CANCELLED = 212
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--username', dest='username', required=True)
+    parser.add_argument('--password', dest='password', required=True)
+    parser.add_argument('--site-id', dest='site_id', required=True)
+    parser.add_argument('--server-url', dest='server_url', required=True)
+    parser.add_argument('--datasource-name', dest='datasource_name', required=True)
+    args = parser.parse_args()
+    return args
+
+
+def authenticate_tableau(username, password, site_id, server_url):
+    """TSC library to sign in and sign out of Tableau Server and Tableau Online.
+
+    :param username:The name of the user.
+    :param password:The password of the user.
+    :param site_id: The site_id for required datasources. ex: ffc7f88a-85a7-48d5-ac03-09ef0a677280
+    :param server_url: This corresponds to the contentUrl attribute in the Tableau REST API.
+    :return: connection object
+    """
+    try:
+        tableau_auth = TSC.TableauAuth(username, password, site_id=site_id)
+        server = TSC.Server(server_url, use_server_version=True)
+        server.auth.sign_in(tableau_auth)
+    except Exception as e:
+        print(f'Failed to connect to Tableau.')
+        print(e)
+        sys.exit(EXIT_CODE_INVALID_CREDENTIALS)
+    return server, tableau_auth
+
+
+def get_datasource(server, tableau_auth, datasource_name):
+    """Returns the specified data source item.
+
+    :param tableau_auth:
+    :param datasourceid: The datasource_item specifies the data source to update.
+    :return: datasource : get the data source item to update
+    """
+    try:
+        datasource_id = None
+        with server.auth.sign_in(tableau_auth):
+            all_datasources, pagination_item = server.datasources.get()
+            for datasources in all_datasources:
+                # print(datasources.name)
+                if datasources.name == datasource_name:
+                    # view_obj = server.datasources.get_by_id(datasourceid)
+                    datasource_id = datasources.id
+                    break
+    except Exception as e:
+        print(f'Datasource item may not be valid or Datasource must be retrieved from server first.')
+        print(e)
+        sys.exit(EXIT_CODE_INVALID_RESOURCE)
+    return datasource_id
+
+
+def refresh_datasource(server, tableau_auth, datasourceobj):
+    """Refreshes the data of the specified extract.
+
+    :param tableau_auth:
+    :param datasourceobj: The datasource_item specifies the data source to update.
+    :return: datasource : datasource object
+    """
+    try:
+        with server.auth.sign_in(tableau_auth):
+            refreshed_datasource = server.datasources.refresh(datasourceobj)
+            print(f'Datasource refresh trigger successful.')
```

Comment:
Let's make sure all of our succes/error messages are super specific by including the datasource_name or other relevant inputs.

In this case `f'Datasource {datasource_name} was successfully triggered.'`

That code won't work directly thought because of the function inputs.